File: code_examples/3-Structured_Problem_Solving/8-If_Then_ElseIf_Else/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module checks a traffic light
"""


def main() -> None:
    """The main() function checks a traffic light, returns None."""

    # input
    light_color = input("Enter the color of the traffic light: ")

    # process and output
    if light_color == "red":
        print("Stop!")
    elif light_color == "yellow":
        print("Slow down!")
    elif light_color == "green":
        print("Go!")
    else:
        print("Not a valid color.")

    print("\nDone.")


if __name__ == "__main__":
    main()

File: code_examples/3-Structured_Problem_Solving/8-If_Then_ElseIf_Else/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program checks a traffic light
 */

package main

import "fmt"

func main() {
	// this function checks a traffic light
	var lightColor string

	// input
	fmt.Print("Enter the color of the traffic light: ")
	fmt.Scan(&lightColor)

	// process and output
	if lightColor == "red" {
		fmt.Println("Stop!")
	} else if lightColor == "yellow" {
		fmt.Println("Slow down!")
	} else if lightColor == "green" {
		fmt.Println("Go!")
	} else {
		fmt.Println("Not a valid color.")
	}

	fmt.Println("\nDone.")
}
